helios-testing/src/main/java/com/spotify/helios/testing/HeliosSoloDeployment.java

```diff
@@ -157,6 +161,22 @@
     }
   }
 
+  /**
+   * Return first found non-loopback ip from enumeration of network devices.
+   * InetAddress.getLocalHost() will not work, as it returns the loopback device
+   */
+  public static String findFirstInet4Addr(Enumeration<NetworkInterface> nics)
+      throws HeliosDeploymentException {
+    for(NetworkInterface nic : Collections.list(nics)) {
+      for(InetAddress addr : Collections.list(nic.getInetAddresses())) {
+        if(addr instanceof Inet4Address && !addr.isLoopbackAddress()) {
```

Comment:
@mattnworb I made the changes you suggested.  One thing to note is that Inet4Address is not mockable, being a final class, nor does it have a public constructor.  As this prevents the possibility of cleanly unit testing, I combined the logic into a single method.

Reply:
This might be helpful for creating stub inetaddress instances in your test: https://github.com/google/guava/blob/08983fb25b9f915d2ca51393697f34c9a3854bbd/guava/src/com/google/common/net/InetAddresses.java#L135